hs_core/models.py

```diff
@@ -1592,23 +1623,27 @@ def get_xml(self, pretty_print=True):
             hsterms_pub_url = etree.SubElement(dc_pub_rdf_Description, '{%s}publisherURL' % self.NAMESPACES['hsterms'])
             hsterms_pub_url.set('{%s}resource' % self.NAMESPACES['rdf'], self.publisher.url)
 
+        res_url_root = '{site_url}/resource/'.format(site_url = current_site_url())
         for rel in self.relations.all():
             dc_relation = etree.SubElement(rdf_Description, '{%s}relation' % self.NAMESPACES['dc'])
             dc_rel_rdf_Description = etree.SubElement(dc_relation, '{%s}Description' % self.NAMESPACES['rdf'])
-            rel_dcterm = '{%s}' + rel.type
-            dcterms_type = etree.SubElement(dc_rel_rdf_Description, rel_dcterm % self.NAMESPACES['dcterms'])
-            # check if the relation value starts with 'http://' or 'https://'
-            if rel.value.lower().find('http://') == 0 or rel.value.lower().find('https://') == 0:
-                dcterms_type.set('{%s}resource' % self.NAMESPACES['rdf'], rel.value)
+            if rel.type in Relation.HS_RELATION_TERMS:
+                term_ns = self.NAMESPACES['hsterms']
             else:
-                dcterms_type.text = rel.value
+                term_ns = self.NAMESPACES['dcterms']
+            terms_type = etree.SubElement(dc_rel_rdf_Description, '{%s}%s' % (term_ns, rel.type))
+
+            if rel.value.lower().find(res_url_root) == 0:
```

Comment:
The old code just checks to see if the value attribute of the relation element is a URI (which can be any valid URI including ones based on hydroshare) and encodes in one specific way (using rdf:resource) otherwise it encodes differently. The new code will encode only hydroshare based URI (using the rdf:resource). I am not sure which one is correct. I am just pointing the difference. @horsburgh can tell us which one is right.